src/main/java/gift/filter/JwtAuthFilter.java

```diff
@@ -0,0 +1,45 @@
+package gift.filter;
+
+import gift.security.JwtProvider;
+import jakarta.servlet.FilterChain;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.web.filter.OncePerRequestFilter;
+
+import java.io.IOException;
+
+public class JwtAuthFilter extends OncePerRequestFilter {
+
+    private final JwtProvider jwtProvider;
+
+    public JwtAuthFilter(JwtProvider jwtProvider) {
+        this.jwtProvider = jwtProvider;
+    }
+
+    @Override
+    protected void doFilterInternal(HttpServletRequest request,
+                                    HttpServletResponse response,
+                                    FilterChain filterChain)
+            throws ServletException, IOException {
+
+        String authHeader = request.getHeader("Authorization");
+
+        if (authHeader == null || !authHeader.startsWith("Bearer ")) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        String token = authHeader.substring(7);
+
+        if (!jwtProvider.validateToken(token)) {
+            response.setStatus(HttpServletResponse.SC_UNAUTHORIZED);
+            return;
+        }
+
+        Long userId = jwtProvider.getUserId(token);
+        request.setAttribute("userId", userId);
+
+        filterChain.doFilter(request, response);
+    }
```

Comment:
- 값을 하드코딩하기 보다, 상수를 활용하는건 어떨까요?
```suggestion
    protected void doFilterInternal(HttpServletRequest request,
                                    HttpServletResponse response,
                                    FilterChain filterChain)
            throws ServletException, IOException {
        
        JwtAuthenticationResult result = authenticationHandler.handle(request);
        
        if (result.isSuccess().not()) {
            responseHandler.handleUnauthorized(response, result.getErrorMessage());
            return;
        }
        ...
```
- 이런 비기능 요구사항도 객체지향 설계해보는건 어떨까요?